File: source_monitor.py
from datetime import datetime, timezone

from flask import Blueprint, jsonify, request


source_monitor_bp = Blueprint(
    "source_monitor",
    __name__,
    url_prefix="/api/release4/sources",
)

_sources = [
    {
        "bank": "Сбер",
        "product": "Вклад",
        "source_url": "https://www.sberbank.ru/",
        "status": "needs_check",
        "checked_at": "не проверено",
    },
    {
        "bank": "Т-Банк",
        "product": "Накопительный счет",
        "source_url": "https://www.tbank.ru/",
        "status": "needs_check",
        "checked_at": "не проверено",
    },
]


@source_monitor_bp.get("")
def list_sources():
    return jsonify({"items": _sources})


@source_monitor_bp.post("/mark")
def mark_source():
    payload = request.get_json(silent=True) or {}
    bank = payload.get("bank")
    status = payload.get("status", "ok")

    for source in _sources:
      if source["bank"] == bank:
          source["status"] = status
          source["checked_at"] = datetime.now(timezone.utc).isoformat()
          return jsonify(source)

    return jsonify({"error": "Source not found"}), 404

File: source_monitor.js
(function () {
  const root = document.querySelector("[data-source-monitor]");

  if (!root) {
    return;
  }

  const list = root.querySelector("[data-source-list]");
  const refreshButton = root.querySelector("[data-source-refresh]");
  const fallbackSources = [
    { bank: "Сбер", product: "Вклад", checked_at: "не проверено", status: "needs_check" },
    { bank: "Т-Банк", product: "Накопительный счет", checked_at: "не проверено", status: "needs_check" },
    { bank: "ВТБ", product: "Вклад", checked_at: "не проверено", status: "needs_check" }
  ];

  function renderSources(items) {
    list.innerHTML = items.map((item) => {
      const isOk = item.status === "ok";
      const label = isOk ? "Актуально" : "Нужно проверить";

      return `
        <article class="source-monitor__card">
          <span class="source-monitor__status ${isOk ? "" : "source-monitor__status--warning"}">${label}</span>
          <h3>${item.bank}</h3>
          <p>${item.product}</p>
          <p>Последняя проверка: ${item.checked_at}</p>
        </article>
      `;
    }).join("");
  }

  async function loadSources() {
    refreshButton.disabled = true;
    refreshButton.textContent = "Проверяем...";

    try {
      const response = await fetch("http://127.0.0.1:5000/api/release4/sources");
      if (!response.ok) {
        throw new Error("Sources request failed");
      }
      const data = await response.json();
      renderSources(data.items || fallbackSources);
    } catch (error) {
      renderSources(fallbackSources);
    } finally {
      refreshButton.disabled = false;
      refreshButton.textContent = "Проверить";
    }
  }

  refreshButton.addEventListener("click", loadSources);
})();
